database/src/main/java/database/service/JsonDatabaseService.java

```diff
@@ -114,6 +114,36 @@ public <T extends BaseEntity> T addNewRecordToTable(T entity) {
         return entity;
     }
 
+    @Override
+    public <T extends BaseEntity> Iterable<T> addNewRecordsToTable(Class<? extends BaseEntity> entityClass,
+                                                                   List<T> entities) {
+        if (entities == null || entities.isEmpty()) {
+            throw new IllegalArgumentException("Entities list cannot be null or empty");
+        }
+
+        Path databasePath = Path.of(getDatabasePath(entityClass));
+
+        if (!Files.exists(databasePath)) {
+            LOG.error(DB_FILE_NOT_EXIST + ": {}", databasePath.toAbsolutePath());
+            throw new DatabaseDoesNotExistException(DB_FILE_NOT_EXIST);
+        }
+
+        List<T> existingEntities = deserializeEntities(entityClass, readDatabaseFile(databasePath));
+        String entityClassName = entityClass.getName();
+
+        for (T entity : entities) {
+            if (entity.getId() != null) {
+                throw new IdProvidedManuallyException(ID_PROVIDED_MANUALLY);
+            }
+            entity.setId(entityIds.merge(entityClassName, 1, Integer::sum));
```

Comment:
You have a lot of common code in `addNewRecordToTable` and `addNewRecordsToTable` methods. Consider creating several common methods for validation and setting id to entities. Even if those methods will be small, it's still more appropriate way to organize your code.